pandas/_libs/lib.pyx

```diff
@@ -2315,3 +2312,119 @@ def fast_multiget(dict mapping, ndarray keys, default=np.nan):
             output[i] = default
 
     return maybe_convert_objects(output)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline object convert_to_unicode(object item,
+                                      bint keep_trivial_numbers):
+    """
+    Convert `item` to str.
+
+    Parameters
+    ----------
+    item : object
+    keep_trivial_numbers : bool
+        if True, then conversion (to string from integer/float zero)
+        is not performed
+
+    Returns
+    -------
+    str
+    """
+    cdef:
+        bint do_convert = 1
+        float64_t float_item
+
+    if keep_trivial_numbers:
+        if isinstance(item, int):
+            if <int>item == 0:
+                do_convert = 0
+        elif isinstance(item, float):
+            float_item = item
+            if float_item == 0.0 or float_item != float_item:
+                do_convert = 0
+
+    if do_convert and not isinstance(item, str):
```

Comment:
when is this *not* a string?

Reply:
For example, when `date_cols` contains several arrays that contain nan objects.

Reply:
Another example would be non-trivial number coming in.
Consider parsing the following csv:

```
20100101,20100102
20100103,20100104
...
```

My understanding would be that first-level parser would parse all items as `int64` ones, and then when we need to parse out dates (as I intentionally used ISO8601 format with no separators) date parsing function expects these as strings.